### PR TITLE
rgbds: 0.6.1 -> 0.7.0

### LINKED ISSUE #283282: Update request: rgbds 0.6.1 → 0.7.0
- Package name: rgbds
- Latest released version: 0.7.0

- Current version on the unstable channel: 0.6.1
- Current version on the stable/release channel: 0.6.1

- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

**Notify maintainers**
@meithecatte @matthewbauer 

-----

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by Taneb**
## Description of changes

Closes #283282

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/compilers/rgbds/default.nix [hunk 1]
```diff
@@ -2,15 +2,18 @@
...
     repo = "rgbds";
     rev = "v${version}";
-    sha256 = "sha256-3mx4yymrOQnP5aJCzPWl5G96WBxt1ixU6tdzhhOsF04=";
+    sha256 = "sha256-aktKJlwXpHpjSFxoz5wZJPGWZIcn4ax5iBP0GQEux78=";
   };
   nativeBuildInputs = [ bison flex pkg-config ];
   buildInputs = [ libpng ];
+  patchPhase = ''
+    patchShebangs --host src/bison.sh
+    '';
```
Comment (drupol) [inline pkgs/development/compilers/rgbds/default.nix:16]:
```suggestion
  postPatch = ''
    patchShebangs --host src/bison.sh
  '';
```

**Review by drupol — CHANGES_REQUESTED** [review]

**Comment by drupol** [comment]
Please squash those commits, there should only be 1 commit.

**Comment by Taneb** [comment]
@drupol done

File: pkgs/development/compilers/rgbds/default.nix [hunk 2]
```diff
@@ -2,15 +2,18 @@
...
 stdenv.mkDerivation rec {
   pname = "rgbds";
-  version = "0.6.1";
+  version = "0.7.0";
   src = fetchFromGitHub {
     owner = "gbdev";
     repo = "rgbds";
     rev = "v${version}";
-    sha256 = "sha256-3mx4yymrOQnP5aJCzPWl5G96WBxt1ixU6tdzhhOsF04=";
+    sha256 = "sha256-aktKJlwXpHpjSFxoz5wZJPGWZIcn4ax5iBP0GQEux78=";
```
Comment (drupol) [inline pkgs/development/compilers/rgbds/default.nix:10]:
```suggestion
    hash = "sha256-aktKJlwXpHpjSFxoz5wZJPGWZIcn4ax5iBP0GQEux78=";
```

**Review by drupol — CHANGES_REQUESTED** [review]

File: pkgs/development/compilers/rgbds/default.nix (hunk 2 above)
Comment (drupol) [reply pkgs/development/compilers/rgbds/default.nix:10]: Sorry I just spotted this right now ^^

Comment (Taneb) [reply pkgs/development/compilers/rgbds/default.nix:10]: I've never seen that pattern before! Fix upcoming...

Comment (drupol) [reply pkgs/development/compilers/rgbds/default.nix]: This is the default since quite a while now.

**Comment by drupol** [comment]
@ofborg build rgbds

**Review by drupol — APPROVED** [review]

**Comment by drupol** [comment]
Merging this since I've got no answer from maintainers.

**Pull request merged**